Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/file_bugs.py

```diff
@@ -0,0 +1,197 @@
+        # FIXME: This is very hardcoded to the exact template, could probably be a bit more
+        # adaptive.
+        desc = []
+        for line in lines[desc_start:]:
+            if line.startswith('* '):
+                break
+            desc.append(line)
+        return "\n".join(desc)
+
+    @classmethod
+    def create_bug(cls, repository, commit):
+        lines = commit.message.splitlines()
+        if len(lines) < 2:
+            return
+
+        issue = None
+        radar = None
+        needs_radar = False
+        if cls.BUGZILLA_PLACEHOLDER not in lines[1]:
+            issue = Tracker.from_string(lines[1])
+            if not issue or not isinstance(issue.tracker, BugzillaTracker):
+                return
+        if len(lines) > 2:
+            needs_radar = cls.RADAR_PLACEHOLDER in lines[2]
+            if not needs_radar:
+                radar = Tracker.from_string(lines[2])
+                if not radar or not isinstance(radar.tracker, RadarTracker):
+                    return
+
+        if issue and (radar or not needs_radar):
+            return
+
+        title = lines[0]
+        print('Found a commit without an issue specified. Title: ' + title)
+        if not User().confirm("Create issue now?", User.DEFAULT_NO):
+            return
+
+        if not issue:
+            if not User().confirm("Use existing title?", User.DEFAULT_YES):
+                title = Terminal.input('Issue title: ')
+            desc = cls.parse_description(lines)
+            print('Parsed description: ' + desc)
+            if not User().confirm("Use existing description?", User.DEFAULT_YES):
+                desc = Terminal.input('Issue description: ')
+
+            # FIXME: This is just blindly assuming bugzilla, but also not actually
+            # confirming that.
+            if getattr(Tracker.instance(), 'credentials', None):
+                Tracker.instance().credentials(required=True, validate=True)
+            issue = Tracker.instance().create(
+                title=title,
+                description=desc,
+            )
+            if not issue:
+                sys.stderr.write('Failed to create new issue\n')
+                return
+            print("Created '{}'".format(issue))
+            if radar:
+                print('Found existing rdar to CC: ' + str(radar))
+                issue.cc_radar(block=True, radar=radar)
+        if needs_radar:
+            radar = issue.cc_radar(block=True, radar=None)
+            print("Created '{}'".format(radar))
+
+        # Try to mutate the commit message and insert our newly filed issue.
+        # Currently, we can only do this for the HEAD command, and if there's no staged
+        # changes that our amend would accidentally bring in. Is this a sufficient check?
+        # Detached HEAD state? Error checking on the amend git command?
+        # Should we pre-warn at the start of the command if we won't be able to write
+        # the results?
```

Comment:
This comment makes me wonder if we're better off just operating on a single commit....otherwise, I think we need to use `git filter-branch`...it's possible, `git-webkit canonicalize` uses `git filter-branch` extensively, but it's a really hard command to use.

Reply:
I think it's worth having the ability to file bugs for a range of commits, even if you have to do the `git rebase -I` manually afterwards to insert them into the commit messages. That's still a lot less manual work than filing the bugs by hand, or moving commits around so that each of them ends up without descendants.